setup-tdx-guest.sh

```diff
@@ -33,20 +35,25 @@ Pin: release o=LP-PPA-kobuk-team-tdx
 Pin-Priority: 4000
 EOF
 
-apt update
+# upgrade the system to have the latest components (mostly generic kernel)
+apt upgrade --yes
 
 # install TDX feature
 # install modules-extra to have tdx_guest module
 apt install --yes --allow-downgrades \
-   linux-image-unsigned-${KERNEL_RELEASE} \
-   linux-modules-extra-${KERNEL_RELEASE} \
    shim-signed \
    grub-efi-amd64-signed \
    grub-efi-amd64-bin \
    tdx-tools-guest \
    python3-pytdxmeasure
 
-grub_switch_kernel ${KERNEL_RELEASE}
+# if a specific kernel has to be used
+# TODO : install linux-modules-extra
+if [ ! -z "${KERNEL_RELEASE}" ]; then
+  apt install --yes --allow-downgrades \
+    linux-image-unsigned-${KERNEL_RELEASE}
+  grub_switch_kernel ${KERNEL_RELEASE}
```

Comment:
Will have to handle the case where user selects the generic kernel and the current system is using other kernel (-intel kernel from previous setup for example)